packages/x-charts/src/ChartsTooltip/ChartsAxisTooltipContent.tsx

```diff
@@ -74,17 +76,25 @@ function ChartsAxisTooltipContent(props: {
           if (axisKey === undefined || axisKey === USED_AXIS_ID) {
             const seriesToAdd = series[seriesType]!.series[seriesId];
 
-            const color = colorGetter(
-              seriesToAdd,
-              xAxis[seriesToAdd.xAxisKey ?? xAxisIds[0]],
-              yAxis[seriesToAdd.yAxisKey ?? yAxisIds[0]],
-            );
-            rep.push({ ...seriesToAdd, getColor: color });
+            const getColor =
+              seriesToAdd.type === 'scatter'
+                ? colorGetter(
+                    seriesToAdd,
+                    xAxis[seriesToAdd.xAxisKey ?? xAxisIds[0]],
+                    yAxis[seriesToAdd.yAxisKey ?? yAxisIds[0]],
+                    zAxis[seriesToAdd.zAxisKey ?? zAxisIds[0]],
+                  )
+                : colorGetter(
+                    seriesToAdd,
+                    xAxis[seriesToAdd.xAxisKey ?? xAxisIds[0]],
+                    yAxis[seriesToAdd.yAxisKey ?? yAxisIds[0]],
+                  );
```

Comment:
Wouldn't simplifying this to just the first part work? Like, if zAxis is null/undefined it would default to the same?

```tsx
colorGetter(
  seriesToAdd,
  xAxis[seriesToAdd.xAxisKey ?? xAxisIds[0]],
  yAxis[seriesToAdd.yAxisKey ?? yAxisIds[0]],
  zAxis[seriesToAdd.zAxisKey ?? zAxisIds[0]],
)
```

An alternative would be 


```tsx
const getColor = colorGetter(
  seriesToAdd,
  xAxis[seriesToAdd.xAxisKey ?? xAxisIds[0]],
  yAxis[seriesToAdd.yAxisKey ?? yAxisIds[0]],
  seriesToAdd.type === 'scatter' 
    ? zAxis[seriesToAdd.zAxisKey ?? zAxisIds[0]]
    : undefined
)
```

Reply:
That's because the typing of the `getColor` is strict.depending on the series provided the number of argument can vary.

I modified this section of code to simplify the switching between different kind of series